### PR TITLE
Add checks for certain variables in veJOE staking

### PR DESCRIPTION
**Pull request by 0xnivek**
This PR adds checks to enforce:

- `boostedThreshold > 0`
- `boostedDuration <= 365 days`
- `maxCap <= 100000`

and also adds tests

### CODE REVIEW

File: contracts/VeJoeStaking.sol [hunk 1]
```diff
@@ -41,6 +41,9 @@ contract VeJoeStaking is Initializable, OwnableUpgradeable {
     /// For example, if user has `n` JOE staked, they can own a maximum of `n * maxCap` veJOE.
     uint256 public maxCap;
 
+    /// @notice The upper limit of `maxCap`
+    uint256 public MAX_MAX_CAP;
```
Comment (LouisMeMyself) [inline contracts/VeJoeStaking.sol:45]: I don’t think this is a good name for a variable..
Maybe `UPPER_BOUND_MAX_CAP` or `UPPER_LIMIT_MAX_CAP` ?

File: contracts/VeJoeStaking.sol [hunk 2]
```diff
@@ -60,6 +63,9 @@ contract VeJoeStaking is Initializable, OwnableUpgradeable {
     /// @notice The length of time a user receives boosted benefits
     uint256 public boostedDuration;
 
+    /// @notice The upper limit of `boostedDuration`
+    uint256 public MAX_BOOSTED_DURATION;
```
Comment (LouisMeMyself) [inline contracts/VeJoeStaking.sol:67]: If we change the name for the previous upper bound, we should rename this too

File: contracts/VeJoeStaking.sol [hunk 3]
```diff
@@ -60,6 +63,9 @@ contract VeJoeStaking is Initializable, OwnableUpgradeable {
     /// @notice The length of time a user receives boosted benefits
     uint256 public boostedDuration;
 
+    /// @notice The upper limit of `boostedDuration`
+    uint256 public UPPER_LIMIT_BOOSTED_DURATION;
```
Comment (jummy123) [inline contracts/VeJoeStaking.sol:67]: One of the feedback I have received before is not to name these variables in UPPERCASE_WITH_UNDERSCORES as they aren't actually `immutable` variables.

File: contracts/VeJoeStaking.sol [hunk 4]
```diff
@@ -87,17 +93,26 @@ contract VeJoeStaking is Initializable, OwnableUpgradeable {
         uint256 _boostedDuration,
         uint256 _maxCap
     ) public initializer {
+        UPPER_LIMIT_BOOSTED_DURATION = 365 days;
```
Comment (jummy123) [inline contracts/VeJoeStaking.sol:96]: Optional, you can define this just before the `require` statements where they are used, saving gas on failed txs.

File: contracts/VeJoeStaking.sol [hunk 5]
```diff
@@ -87,17 +93,26 @@ contract VeJoeStaking is Initializable, OwnableUpgradeable {
...
+        UPPER_LIMIT_MAX_CAP = 100000;
```
Comment (cryptofish7) [inline contracts/VeJoeStaking.sol:97]: Can we define `UPPER_LIMIT_MAX_CAP`, `UPPER_LIMIT_BOOSTED_DURATION` and `PRECISION` as immutable variables and assign them in a constructor instead?

Or, we don't use upper snake case.

**Review by cryptofish7 — CHANGES_REQUESTED** [review]

**Review by cryptofish7 — APPROVED** [review]

**Pull request merged**